samples/containers/android/src/main/java/com/squareup/sample/container/panel/PanelOverlayDialogFactory.kt

```diff
@@ -0,0 +1,50 @@
+package com.squareup.sample.container.panel
+
+import android.app.Dialog
+import android.graphics.drawable.ColorDrawable
+import android.util.TypedValue
+import android.view.View
+import com.squareup.sample.container.R
+import com.squareup.workflow1.ui.WorkflowUiExperimentalApi
+import com.squareup.workflow1.ui.container.Bounds
+import com.squareup.workflow1.ui.container.ModalScreenOverlayDialogFactory
+import com.squareup.workflow1.ui.container.maintainBounds
+import com.squareup.workflow1.ui.container.setBounds
+
+@OptIn(WorkflowUiExperimentalApi::class)
+internal object PanelOverlayDialogFactory : ModalScreenOverlayDialogFactory<PanelOverlay<*>>(
+  type = PanelOverlay::class
+) {
+  override fun buildDialogWithContentView(contentView: View): Dialog {
+    val context = contentView.context
+    return Dialog(context, R.style.PanelDialog).also { dialog ->
+      dialog.setContentView(contentView)
+
+      val typedValue = TypedValue()
+      context.theme.resolveAttribute(android.R.attr.windowBackground, typedValue, true)
+      if (typedValue.type in TypedValue.TYPE_FIRST_COLOR_INT..TypedValue.TYPE_LAST_COLOR_INT) {
+        dialog.window!!.setBackgroundDrawable(ColorDrawable(typedValue.data))
+      }
+
+      dialog.maintainBounds(contentView) { _, bounds ->
```

Comment:
Quick thought: what do you think of separating positioning from creation?
Here you are:
1) creating the dialog
2) positioning it for the first time
3) maintaining positions

How to do that?
Paraphrasing Market: the thing that controls the dialog (in Market the runner)i
- implements buildDialog
- receives the values (or flow) for cover area, anchor view and what not.
A probably base code would react to any of those changes by doing:
{ if dialog showing -> updateDialogLocation }
- concrete runner/controller class would only override updateDialogLocation.
In that case: you're not leveraging the great maintain bounds below.
But, you could still do:
maintainBounds(coverAreaFlow) { updateDialogLocation }
maintainBounds(anchorAreaFlow) { updateDialogLocation }

Reply:
I'd still want to pass the `StateFlow<Bounds>` through the `ViewEnvironment`, so that that whole notion of positioning doesn't have to be built into `LayeredDialogs` and `OverlayDialogFactory` -- different containers can create different policies.

But the more specialized `ModalScreenOverlayDialogFactory` could certainly offer an `abstract fun updateBounds(bounds: Bounds)`, would be more foolproof. And our app can build an `AnchorOverlayDialogFactory` for that use case, etc.

Sold.

Reply:
Probably still use `Dialog.maintainBounds` to implement that, but now it doesn't have to be public. I like that.